src/service.ts

```diff
@@ -153,8 +153,26 @@ function parseCLIArguments(args: string[]): [CLIArguments, string, CliOptions] {
           parsedArguments.ignorePath = nextArg.value;
           break;
         }
+        case "--stdin-filepath": {
+          const nextArg = argsIterator.next();
+          if (nextArg.done) {
+            throw new Error("--stdin-filepath option expects a file path");
+          }
+
+          fileName = nextArg.value;
+          break;
+        }
         default: {
-          optionArgs.push(arg);
+          if (arg.includes("=")) {
+            optionArgs.push(arg);
+          } else {
+            const nextArg = argsIterator.next();
+            if (nextArg.done) {
+              throw new Error(`--${arg} expects a value`);
+            }
```

Comment:
I think the problem is that there are cases where this isn't true (some examples include `--bracket-same-line` and `--single-quote`), so it's hard to solve. We may need to actually use something that has the specific options and try to keep them in sync with prettier, which is a pain :(

Reply:
I was playing around in the repl and got this out:

```js
(await require("prettier").getSupportInfo()).options.filter(o => o.type === "boolean").map(o => o.name)
```

```js
[
  'singleQuote',
  'bracketSpacing',
  'bracketSameLine',
  'singleAttributePerLine',
  'vueIndentScriptAndStyle',
  'semi',
  'experimentalTernaries',
  'jsxSingleQuote',
  'insertPragma',
  'requirePragma',
  'useTabs'
]
```

These are all the prettier _options_[^1] that don't take an argument (some have `o.default: true`, meaning the argument would have a _no-_ prefix, like `--no-semi`).

This seems like something that can be worked with, but the arguments could be different depending on the resolved prettier version, which wouldn't be a problem, except that we don't know the version until we've parsed the arguments :^).

Given how prettierd doesn't support `prettierd file.ts`, I'm more inclined to suggest that `--stdin-filepath <filename>` should be required, which would be a breaking change, but far simpler.

[^1]: the prettier CLI has other arguments that don't take a value, but here we're calling the module

Reply:
Hmm yeah I think that's an OK breaking change to make, as it simplifies things and makes prettierd command line interface compatible with prettier.

Do you want to give it a shot?

Reply:
Sure, will do!